setup.py

```diff
@@ -18,14 +18,14 @@
 from cryptography.x509.oid import NameOID
 from cryptography.hazmat.primitives import hashes
 from cryptography.x509 import Name, CertificateSigningRequestBuilder, NameAttribute, SignatureAlgorithmOID, ExtensionType, OID_TIME_STAMPING, OID_CODE_SIGNING, OID_KEY_USAGE, OID_EXTENDED_KEY_USAGE, OID_BASIC_CONSTRAINTS
-from cryptography.hazmat.primitives.asymmetric import rsa
+from cryptography.hazmat.primitives.asymmetric import ec
 from pathlib import Path
 import json
 
 
 start_marker = '-----BEGIN CERTIFICATE REQUEST-----'
 end_marker = '-----END CERTIFICATE REQUEST-----'
-hash_alg = 'RSASSA_PKCS1_V1_5_SHA_256'
+hash_alg = 'ECDSA_SHA_256'
 sign_oid = '1.2.840.113549.1.1.11'
```

Comment:
I think this should probably be 1.2.840 .10045 .2 .1 (EC_PUBLICKEY_OID)

Reply:
Later on in this file, I use `1.2.840.10045.4.3.2`, what's the difference between that and the ID you've specified, and which should be preferred?

Reply:
1.2.840.10045.4.3.2 is the signature type (ECDSA_WITH_SHA256_OID).  The other is the type of key (EC_PUBLICKEY_OID)